greengrass_mqtt_base_block.py

```diff
@@ -39,22 +39,43 @@ def __init__(self):
 
     def configure(self, context):
         """set up MQTT client properties"""
-        self.client(self.client_id(), useWebsocket=self.use_websocket())
-        self.client.configureEndpoint(self.mqtt_host(), self.mqtt_port())
+        super().configure(context)
+
+        self.client = self.client(self.client_id(),
+                                  useWebsocket=self.use_websocket())
+        self.client.configureEndpoint(hostName=self.mqtt_host(),
+                                      portNumber=self.mqtt_port())
+
+        if self.use_websocket():
+            # only need to configure the root CA and IAM credentials for
+            # websockets
+            self.client.configureIAMCredentials(self.creds().access_key_id(),
+                                                self.creds().secret_key())
+            self.client.configureCredentials(
+                CAFilePath=self.root_ca_path().file)
+        else:
+            # don't need to configure IAM credentials for non-websocket use
+            self.client.configureCredentials(
+                CAFilePath=self.root_ca_path().file,
+                KeyPath=self.private_key_path().file,
+                CertificatePath=self.cert_path().file)
+
         self.client.configureOfflinePublishQueueing(queueSize=-1)
         self.client.configureConnectDisconnectTimeout(self.connect_timeout())
-        self.client.configureCredentials(CAFilePath=self.root_ca_path(),
-                                         KeyPath=self.private_key_path(),
-                                         CertificatePath=self.cert_path())
+        self.client.configureDrainingFrequency(2)
+        self.client.configureMQTTOperationTimeout(5)
+        self.client.configureAutoReconnectBackoffTime(1, 32, 20)
+
         self.connect()
-        super().configure(context)
 
     def stop(self):
         self.disconnect()
```

Comment:
Does this need to be `self.client.disconnect()` rather than just `self.disconnect()`?